🐛 Fixed default content CTA message for page vs post

rshbhgrg opened the pull request:
closes https://github.com/TryGhost/Team/issues/1898

- the default content cta always used the terminology as `post` when showing message that users don't have access to some content
- this caused confusion when users were looking at a page and message showed "This post is for subscribers only"
- updates the message to correctly reflect `page` vs `post` on the default cta
- updates default cta to use accent color from global site property instead of local value(which is same)

rshbhgrg committed:
🐛 Fixed default content CTA message to reflect page vs post

closes https://github.com/TryGhost/Team/issues/1898

- the default content cta always used the terminology as `post` when showing message that users don't have access to some content
- this caused confusion when users were looking at a page and message showed "This post is for subscribers only"
- updates the message to correctly reflect `page` vs `post` on the default cta

rishabhgrg commented on ghost/core/core/frontend/helpers/content.js:
```diff
@@ -19,7 +19,8 @@ function restrictedCta(options) {
     _.merge(this, {
-        accentColor: (options.data.site && options.data.site.accent_color)
+        accentColor: (options.data.site && options.data.site.accent_color),
+        resource: options.data.root?.page ? 'page' : 'post'
```
@ErisDS Probably using `#is` helper in the CTA to render `page` vs `post` conditionally instead of this pattern is better here ? Was originally inclined towards this as it gives easy way for custom content helpers to render `page` vs `post`, but realized later the use `#is` here is more consistent without adding any new variables. Wdyt ?

rishabhgrg replied:
Updated. ✅

ErisDS commented on ghost/core/core/frontend/helpers/tpl/content-cta.hbs:
```diff
@@ -2,13 +2,13 @@
 <aside class="gh-post-upgrade-cta">
     <div class="gh-post-upgrade-cta-content" style="background-color: {{accentColor}}">
```
Not sure when this happened, but IMO this should always be `{{@site.accentColor}}`. It's a global property, there's no reason to extend the theme API here and a local copy?

rishabhgrg replied:
Will still keep the local copy of `accentColor` property for themes as otherwise it will be a breaking change for theme devs already relying on that in custom cta helpers, and we can get rid of that in 5.0.

ErisDS commented on ghost/core/core/frontend/helpers/tpl/content-cta.hbs:
```diff
@@ -2,13 +2,13 @@
     <div class="gh-post-upgrade-cta-content" style="background-color: {{accentColor}}">
         {{#has visibility="paid"}}
-            <h2>This post is for paying subscribers only</h2>
+            <h2>This {{resource}} is for paying subscribers only</h2>
```
We already have the `{{#is}}` helper to determine context of whether something is a post or page. Is there a reason why we can't use the existing pattern here?

rishabhgrg replied:
[Updated](https://github.com/TryGhost/Ghost/pull/15471#discussion_r979900173) ✅

rshbhgrg committed:
Updated content cta to use global accent color property

refs https://github.com/TryGhost/Ghost/pull/15471#discussion_r979902374

- the accent color value used by default content cta was copying the global site property which is redundant, and can be directly used
- originally, the accentColor property was extended to allow a fallback value for content ctas, but was later removed as we added default value to global site property directly
- the accentColor property is now deprecated and will be removed in next version, as existing themes might be relying on it for custom cta helpers

ErisDS commented on ghost/core/core/frontend/helpers/tpl/content-cta.hbs:
```diff
@@ -1,19 +1,31 @@
+    <div class="gh-post-upgrade-cta-content" style="background-color: {{@site.accent_color}}">
         {{#has visibility="paid"}}
-            <h2>This post is for paying subscribers only</h2>
+            {{#is "page"}}
```
Is there a reason for not doing this inline? e.g. `<h2>This {{#is "page"}}page{{else}}post{{/is}} is for paying subscribers only</h2>`

rishabhgrg replied:
Will switch to this, no reason other than this one felt cleaner to me, but no strong prefs. :)

ErisDS commented on ghost/core/core/frontend/helpers/content.js:
```diff
@@ -19,7 +19,8 @@ function restrictedCta(options) {
     options.data = options.data || {};
     _.merge(this, {
-        accentColor: (options.data.site && options.data.site.accent_color)
+        accentColor: (options.data.site && options.data.site.accent_color),
```
Although we can't remove this line with the given changes, we should add a comment above it something like: `// @deprecated in Ghost 5.16.1 never documented & removed from core templates`

ErisDS reviewed (approved):
We need to add a deprecation notice but otherwise this is GTG 👍
